tests/test_supported_models.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+
+from templates.supported_models import MODEL_CONFIGS, ModelConfig, get_supported_models
+
+
+@pytest.fixture
+def sample_azure_deployment():
+    return {"type": "azure", "model_name": "gpt-35-turbo", "deployment_name": "azure"}
+
+
+@pytest.fixture
+def sample_openai_deployment():
+    return {"type": "openai", "model_name": "gpt-3.5-turbo", "deployment_name": ""}
+
+
+def test_model_config_structure():
+    assert isinstance(MODEL_CONFIGS, dict)
+
+    for model_name, config in MODEL_CONFIGS.items():
```

Comment:
Should we also assert the type of the config here? We're not really enforcing it anywhere that it's either a `dict` or a `ModelConfig`, so we could either add an assertion here or an else case raising an error!
```suggestion
    for model_name, config in MODEL_CONFIGS.items():
        assert isinstance(
            config, (dict, ModelConfig)
        ), "Model configuration entry can either be a 'dict' or a 'ModelConfig' object."
```